packages/components/src/components/carousel/carousel.jsx

```diff
@@ -0,0 +1,145 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Swipeable } from 'react-swipeable';
+import Card from './carousel-card.jsx';
+import Nav from './carousel-nav.jsx';
+import Icon from '../icon';
+import { useInterval } from '../../hooks';
+
+const Carousel = ({
+    bullet_color,
+    active_bullet_color,
+    className,
+    initial_index,
+    list,
+    bullet_position,
+    nav_position,
+    show_bullet,
+    show_nav,
+    onItemSelect,
+    autoplay_time,
+    width,
+}) => {
+    const [active_index, setActiveIndex] = React.useState(initial_index);
+
+    const handleNextClick = () => {
+        const next_idx = active_index + 1;
+        const has_reached_end = next_idx === list.length;
```

Comment:
Can we define a variable like `last_slide = list.length`, for better code readability?

Reply:
It only going to be used one line later and only going to be used in one place.
IMO, I think it is not enough for adding a new variable and it adds unnecessary complexity